File: backend/models/exercise.py
from enum import Enum
from sqlalchemy import Column, Integer, String, Boolean, Enum as SQLEnum, Index, func
from backend.database import Base

class MuscleGroupEnum(str, Enum):
    CHEST = "Chest"
    BACK = "Back"
    LEGS = "Legs"
    SHOULDERS = "Shoulders"
    ARMS = "Arms"
    CORE = "Core"
    CARDIO = "Cardio"

class Exercise(Base):
    __tablename__ = "exercises"
    __table_args__ = (
        Index('ix_exercise_name_lower', func.lower('name'), unique=True),
    )
    id = Column(Integer, primary_key= True, index=True)
    name = Column(String(100), nullable=False, index=True)
    muscle_group = Column(SQLEnum(MuscleGroupEnum), nullable = False, index = True)
    is_cardio = Column(Boolean, nullable = False, default = False)
    is_active = Column(Boolean, nullable = False, default = True)

File: frontend/src/schemas/exercise.ts
import { z } from 'zod'

export const MuscleGroupSchema = z.enum([
  'Chest',
  'Back',
  'Legs',
  'Shoulders',
  'Arms',
  'Core',
  'Cardio',
])
export type MuscleGroup = z.infer<typeof MuscleGroupSchema>

export const ExerciseSchema = z.object({
  id: z.number(),
  name: z.string(),
  muscle_group: MuscleGroupSchema,
  is_cardio: z.boolean(),
  is_active: z.boolean(),
})
export type Exercise = z.infer<typeof ExerciseSchema>

export const ExerciseCreateSchema = z.object({
  name: z.string(),
  muscle_group: MuscleGroupSchema,
  is_cardio: z.boolean(),
})
export type ExerciseCreate = z.infer<typeof ExerciseCreateSchema>
